### ISSUE TITLE
payinvoice stuck

### ISSUE THREAD

**Issue by Kixunil** [lightningnetwork/lnd#902]
I opened a channel with Yalls few days ago (saturday), used it couple of times, then left it. Today I attempted to pay to Yalls again and payinvoice command gets stuck.

lnd prints out `revocation window exhausted, unable to send: 3` and it seems to be related, but I'm not 100% sure.

Any ideas?

**Comment by Roasbeef** [comment]
I think the version of y'alls might be running an older version and hasn't yet been updated to implement some stricter checks that we implement when accepting commitments. A short term mitigation is to close out that channel forcibly.

**Comment by Kixunil** [comment]
Hmm can't I close it cooperatively?

**Comment by Roasbeef** [comment]
If the state has desynchronized, then both nodes have different view of what a valid state transition is. As a result, you'll need to force close it.

**Comment by Kixunil** [comment]
Hmm it seems I managed to close it cooperatively (I just wanted to try to see what happens). Anyway, thanks!

BTW `pendingchannels` shows strange output:

```
                "capacity": "100000",
                "local_balance": "96795",
                "remote_balance": "0"
```

0 + 96795 != 100000...

**Comment by Roasbeef** [comment]
A closed channel pays some feeds for the closing transaction.

**Comment by Kixunil** [comment]
Interesting, so that would mean Yalls effectively received nothing?

**Comment by Roasbeef** [comment]
If you never sent anything to them, then yes.

**Comment by Kixunil** [comment]
I sent to them 2300 satoshis, if I count correctly.

**Comment by Roasbeef** [comment]
If the payment didn't go through, it was an idle HTLC. Perhaps we can continue this conversation on IRC or the dev slack?

**Comment by Kixunil** [comment]
Sorry, I'm a bit tired and drunk now, so I'll keep this tab open and try to contact you later. I expect to be busy at least until Saturday evening (UTC+1), but I'm not really sure.

Last thing I'm going to say over this channel for now is I paid Yalls and they showed me the articles, so it seems they received my payments. But maybe I'm missing something.

**Comment by Roasbeef** [comment]
Closing for now, will re-open if this issue re-arises after #920 is merged.

**Comment by Kixunil** [comment]
Still the same problem. (Yes, I did upgrade to 4.1.)

To be precise it at first timed out with "payment attempt not completed before timeout of 1m0s", the second attempt is stuck. I don't see any interesting message in lnd output.

**Comment by cfromknecht** [comment]
@Kixunil the issue you're describing should be fixed when #1019 is merged

**Comment by Kixunil** [comment]
Great, thanks for heads up!
